### DISCUSSION TITLE
HeadlessUI Listbox integration.

### DISCUSSION THREAD

**Discussion by lvegerano** [TanStack/virtual#530] [Q&A] [answered]
Has anyone been able to implement this with headlessui/listbox? I've bee trying for a while and have not been able to get it to work. [Here](https://stackblitz.com/edit/github-sizhvg?file=pages%2Findex.tsx) I have an example of their infinite scroll works when you initially open the listbox but breaks after closing it. Any pointers would be greatly appreciated.

**Comment by piecyk** [comment, marked as answer]
@lvegerano issue here is that the scroll element changes as the Options unmounts, one option would be to store the scroll element on state, something like https://stackblitz.com/edit/github-sizhvg-gxxtjm

**Reply by lvegerano** [reply]
Thank you. I knew it would be something silly :)

**Reply by p1pah** [reply]
funny finding you here @lvegerano I was having the same issue 🤣
